ui/app/AppMain.qml

```diff
@@ -352,44 +352,42 @@ Item {
-                    {
-                        var obj = this.children[i];
-                        if(obj && obj.sectionId == mainModule.activeSection.id)
-                        {
-                            return i
-                        }
+                commLoader.active = false;
+                if (mainModule.activeSection.sectionType === Constants.appSection.chat) {
+
+                    /*************************************/
+                    // This will be refactored later
+                    if (chatsModel.communities.activeCommunity.active) {
+                        chatLayoutContainer.chatColumn.hideChatInputExtendedArea();
+                        chatsModel.communities.activeCommunity.active = false
                     }
+                    /*************************************/
 
-                    // Should never be here, correct index must be returned from the for loop above
-                    console.error("Wrong section type: ", mainModule.activeSection.sectionType,
-                                  " or section id: ", mainModule.activeSection.id)
-                    return Constants.appViewStackIndex.community
+                    return 0;
+                } else if (mainModule.activeSection.sectionType === Constants.appSection.community) {
+                    chatsModel.communities.setActiveCommunity(mainModule.activeSection.id);
+                    mainModule.prepareCommunitySectionModuleForCommunityId(mainModule.activeSection.id);
+                    commLoader.active = true;
+                    return 1;
                 }
                 else if(mainModule.activeSection.sectionType === Constants.appSection.wallet) {
-                    return Constants.appViewStackIndex.wallet
+                    return 2;
                 }
                 else if(mainModule.activeSection.sectionType === Constants.appSection.walletv2) {
-                    return Constants.appViewStackIndex.walletv2
+                    return 3;
                 }
                 else if(mainModule.activeSection.sectionType === Constants.appSection.browser) {
-                    return Constants.appViewStackIndex.browser
+                    return 4;
                 }
                 else if(mainModule.activeSection.sectionType === Constants.appSection.timeline) {
-                    return Constants.appViewStackIndex.timeline
-                }
-                else if(mainModule.activeSection.sectionType === Constants.appSection.profile) {
-                    return Constants.appViewStackIndex.profile
+                    return 5;
                 }
                 else if(mainModule.activeSection.sectionType === Constants.appSection.node) {
-                    return Constants.appViewStackIndex.node
+                    return 6;
+                }
+                else if(mainModule.activeSection.sectionType === Constants.appSection.profile) {
+                    return 7;
```

Comment:
I'd appreciate if we'd re-introduce an equivalent to `Constants.appViewStackIndex.node` etc

Reply:
> 

done